docs/dev/adding_instruments.rst

```diff
@@ -732,6 +644,102 @@ In the above example, :code:`MultimeterA` and :code:`MultimeterB` use a differen
 :code:`MultimeterB` can be defined subclassing :code:`MultimeterA` and just implementing the difference.
 
 
+.. _advanced_communication_protocols:
+
+Advanced communication protocols
+================================
+
+Some devices require a more advanced communication protocol, e.g. due to checksums or device addresses. In most cases, it is sufficient to subclass :meth:`Instrument.write <pymeasure.instruments.Instrument.write>` and :meth:`Instrument.read <pymeasure.instruments.Instrument.read>`.
+
+
+Instrument's inner workings
+***************************
+
+In order to adjust an instrument for more complicated protocols, it is key to understand the different parts.
+
+The :class:`~pymeasure.adapters.Adapter` exposes :meth:`~pymeasure.adapters.Adapter.write` and :meth:`~pymeasure.adapters.Adapter.read` for strings, :meth:`~pymeasure.adapters.Adapter.write_bytes` and :meth:`~pymeasure.adapters.Adapter.read_bytes` for bytes messages, and :meth:`~pymeasure.adapters.Adapter.write_binary_values` and :meth:`~pymeasure.adapters.Adapter.read_binary_values` for binary data like waveforms. Unless you deal with binary data, you do not have to call the Adapter's methods. You should use instead the methods of :class:`~pymeasure.instruments.Instrument` with the same name. They call the Adapter for you and keep the code tidy.
+
```

Comment:
> Unless you deal with binary data

How should one proceed if one _has_ to deal with binary data? (Point to existing instruments as "template"?)

Reply:
I reformulate, that it is allowed to use the binary_values methods and for the other methods, you should choose the Instrument's ones.